cmd/storage/filesystem.go

```diff
@@ -131,6 +132,20 @@ func (f *Filesystem) createPartitions() error {
 				f.log.Error("sgdisk creating partitions failed", "error", err)
 				return fmt.Errorf("unable to create partitions on %s %w", *disk.Device, err)
 			}
+
+			// force the kernel to re-read the partition table
+			osFile, err := gos.OpenFile(*disk.Device, gos.O_RDONLY, 0)
+			if err != nil {
+				return fmt.Errorf("failed to open device %s: %v", *disk.Device, err)
+			}
+			defer osFile.Close()
+
+			fd := osFile.Fd()
+
+			err = unix.IoctlSetInt(int(fd), unix.BLKRRPART, 0)
+			if err != nil {
+				return fmt.Errorf("unable to re-read the partition table. Kernel still uses old partition table: %v", err)
+			}
```

Comment:
Sorry, to mention this now, but there is already a implementation of this available with u-root which we use quite a bit in our code:

https://github.com/u-root/u-root/blob/main/pkg/mount/block/blockdev_linux.go#L183


Should be something like:

```go

blkdev,err := block.Device(*disk.Device)
// Error handling

err = blkdev.ReadPartitionTable()


```